profiler/options.go

```diff
@@ -68,6 +74,17 @@ func (c *config) addProfileType(t ProfileType) {
 	c.types[t] = struct{}{}
 }
 
+func (c *config) skippingAgent() bool {
+	return c.apiKey != ""
+}
+
+func (c *config) targetURL() string {
+	if c.skippingAgent() {
+		return c.apiURL
+	}
+	return c.agentURL
+}
```

Comment:
Maybe this might make more sense to read and also with regards to the new `WithAPIKey` documentation.
```suggestion
func (c *config) skippingAgent() bool {
	return c.targetURL() == c.apiURL
}

func (c *config) targetURL() string {
	if c.apiKey != "" {
		// connect to intake directly
		return c.apiURL
	}
	return c.agentURL
}
```
We could even go one step further and drop `skippingAgent` because `c.targetURL() == c.apiURL` is basically spelling out what's going on.